pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -361,17 +361,42 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(ctx context.Context, cl
 				return nil
 			}
 
+			// The claim can't be bound to any existing volume. let's check if it can be
+			// dynamically provisioned. If can't, we will emit an event to inform the user
+			// the reason why the claim is pending. otherwise, we will provision a volume
+			// for the claim in an asynchronous goroutine and return immediately.
 			switch {
 			case delayBinding && !storagehelpers.IsDelayBindingProvisioning(claim):
+				// If the claim is in delay binding mode and not annotated with
+				// "volume.kubernetes.io/selected-node" by the scheduler's VolumeBinding
+				// plugin, it means that no active pod is observed by the scheduler to
+				// be using this claim. In this case, we should emit an event to inform
+				// the user that the claim is waiting for a pod to be scheduled.
+				//
+				// If the annotation is present, it means that the scheduler has already
+				// assumed a node for the pod and the claim is in delay binding provisioning
+				// mode. In this case, the branch won't be hit.
```

Comment:
That belongs to a branch that handles the case with the annotation present, i.e. `default`.